### PR TITLE
fix(Exception): add basic object exceptions

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: src/geode/basic/assert.cpp
Comment (BotellaA) [inline src/geode/basic/assert.cpp:1]: All abseil exceptions seem to be directly implemented as std::exceptions

File: include/geode/geometry/basic_objects/segment.hpp [hunk 1]
```diff
@@ -102,4 +103,41 @@ namespace geode
...
+// NOLINTNEXTLINE
+#define OPENGEODE_SEGMENT_EXCEPTION( dimension, condition, segment, ... )      \
+    if( ABSL_PREDICT_FALSE( !( condition ) ) )                                 \
+        throw geode::OpenGeodeSegmentException< dimension >                    \
+        {                                                                      \
+            segment, __VA_ARGS__                                               \
+        }
+
+// NOLINTNEXTLINE
+#define OPENGEODE_SEGMENT_EXCEPTION1D( condition, segment, ... )               \
```
Comment (panquez) [inline include/geode/geometry/basic_objects/segment.hpp:134]: Why not put the 1D/2D/3D right after "SEGMENT"?

File: src/geode/basic/assert.cpp
Comment (BotellaA) [reply src/geode/basic/assert.cpp:1]: There is no absl::exception

**Review by MelchiorSchuh — APPROVED** [review]

**Review by panquez — APPROVED** [review]

**Pull request merged**